src/logistration/RegistrationPage.jsx

```diff
@@ -64,10 +93,16 @@ class RegistrationPage extends React.Component {
       email: this.state.email,
       username: this.state.username,
       password: this.state.password,
-      name: this.state.fullname,
-      honor_code: true,
-      country: this.state.country,
+      name: this.state.name,
     };
+
+    const fieldMap = { ...REGISTRATION_VALIDITY_MAP, ...REGISTRATION_OPTIONAL_MAP };
+    Object.keys(fieldMap).forEach((key) => {
+      const value = fieldMap[key];
+      if (value) {
+        payload[key] = this.state[camelCase(key)];
```

Comment:
why are we not getting the camel cased keys directly inside the service when we get the data?

Reply:
If the question is that we should be handling the object directly. The issue is camelCase here is needed in value rather than for keys. So had to be used directly.

Reply:
okay

Reply:
While the above answer does hold for when we are not converting api data keys into camelCase, in this specific case, I am does using camelCase on key. The thing is that i have creating a mapping/dictionary ie fieldMap  which is used to filter out fields based on their name/id which are not in camelCase. I think we can improve on it. Making this mapping in camelCase we might remove at least this direct call all together.

Reply:
yeah